### DISCUSSION TITLE
Issues: associate to Nodes?

### DISCUSSION THREAD

**Discussion by alessiostalla** [Strumenta/StarLasu#32] [Ideas]
Currently, an Issue has a reference to a Position (or Range in Kolasu 1.6).
In some of our transpilers, we create Issues starting from a problematic AST Node. However the issue can't reference the node and it has to store its position instead. If we later want to retrieve the node, or if we want to flag/decorate the node in a viewer with the associated issues, we have to search the AST by position, for each issue.

Would it make sense to do one or both of the following?
1. allow Issue to maintain a reference to "something that has a position" (that could be a Node or just a Position) instead of storing just the Position/Range directly
2. when annotations on nodes are available, also attach Issues to the relevant Node using an annotation

Either point would allow to obtain the issues related to a node without searching them by position. With 1. we make it easy to go from Issue to Node, with 2. the opposite direction from Node to Issues.

**Comment by ftomassetti** [comment]
I think it would make sense to associate issues to nodes, but some issues are not associated to nodes (e.g., lexical and parsing issues). So, yes, the target could either be a Node or a Position/Range.

I think both solutions would make sense.
